File: backend/src/main/java/com/d4k/ecommerce/modules/cart/service/CartService.java
package com.d4k.ecommerce.modules.cart.service;

import com.d4k.ecommerce.modules.cart.dto.request.AddToCartRequest;
import com.d4k.ecommerce.modules.cart.dto.request.UpdateCartItemRequest;
import com.d4k.ecommerce.modules.cart.dto.response.CartResponse;

/**
 * Cart Service Interface
 * Định nghĩa business logic cho giỏ hàng
 */
public interface CartService {
    
    /**
     * Lấy giỏ hàng của user
     * @param userId user ID
     * @return thông tin giỏ hàng
     */
    CartResponse getCart(Long userId);
    
    /**
     * Thêm sản phẩm vào giỏ hàng
     * @param userId user ID
     * @param request thông tin product và quantity
     * @return giỏ hàng sau khi thêm
     */
    CartResponse addToCart(Long userId, AddToCartRequest request);
    
    /**
     * Cập nhật số lượng cart item
     * @param userId user ID
     * @param itemId cart item ID
     * @param request quantity mới
     * @return giỏ hàng sau khi cập nhật
     */
    CartResponse updateCartItem(Long userId, Long itemId, UpdateCartItemRequest request);
    
    /**
     * Xóa item khỏi giỏ hàng
     * @param userId user ID
     * @param itemId cart item ID
     * @return giỏ hàng sau khi xóa
     */
    CartResponse removeCartItem(Long userId, Long itemId);
    
    /**
     * Xóa toàn bộ giỏ hàng (clear cart)
     * @param userId user ID
     */
    void clearCart(Long userId);
}

File: frontend/src/services/cart-service.js
import apiClient from './api-client';

/**
 * Cart Service
 * Các API liên quan đến shopping cart
 */
const cartService = {
  /**
   * Lấy giỏ hàng của user
   * @returns {Promise}
   */
  getCart: () => {
    return apiClient.get('/cart');
  },

  /**
   * Thêm sản phẩm vào giỏ
   * @param {Object} data - { productId, quantity }
   * @returns {Promise}
   */
  addToCart: (data) => {
    return apiClient.post('/cart/add', data);
  },

  /**
   * Cập nhật quantity của item
   * @param {Number} itemId - Cart item ID
   * @param {Object} data - { quantity }
   * @returns {Promise}
   */
  updateCartItem: (itemId, data) => {
    return apiClient.put(`/cart/update/${itemId}`, data);
  },

  /**
   * Xóa item khỏi giỏ
   * @param {Number} itemId - Cart item ID
   * @returns {Promise}
   */
  removeCartItem: (itemId) => {
    return apiClient.delete(`/cart/remove/${itemId}`);
  },

  /**
   * Xóa toàn bộ giỏ hàng
   * @returns {Promise}
   */
  clearCart: () => {
    return apiClient.delete('/cart/clear');
  },
};

export default cartService;

